### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by AndroiableDroid**

### CODE REVIEW

**Comment by AndroiableDroid** [comment]
Still wip don't merge for a while

**Comment by AndroiableDroid** [comment]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,5 @@ tmp/
 out/
 *.pyc
 __pycache__
+utils/oppo_*
+utils/vmlinux*
```
Comment (rokibhasansagar) [inline .gitignore:11]: Thanks for this fix, totally forgot to add these directories into .gitignore :thinking:

File: dumper.sh [hunk 2]
```diff
@@ -513,7 +539,7 @@ elif 7z l -ba "${FILEPATH}" | grep ".*.rar\|.*.zip\|.*.7z\|.*.tar$" || [[ $(find
 		7z e -y "${FILEPATH}" -o"${TMPDIR}"/"${UNZIP_DIR}" 2>/dev/null >> "${TMPDIR}"/zip.log
 		for f in "${TMPDIR}"/"${UNZIP_DIR}"/*; do detox -r "${UNZIP_DIR}"/"${f}" 2>/dev/null; done
 	fi
-	zip_list=$(find ./"${UNZIP_DIR}" -type f -size +300M \( -name "*.rar" -o -name "*.zip" -o -name "*.7z" -o -name "*.tar" \) | cut -d'/' -f'2-' | sort)
+	zip_list=$(find ./"${UNZIP_DIR}" -type f -size +300M \( -name "*.rar" -o -name "*.zip" -o -name "*.7z" -o -name "*.tar"\) | cut -d'/' -f'2-' | sort)
```
Comment (rokibhasansagar) [inline dumper.sh:542]: This is not needed though.., but okay.

File: dumper.sh [hunk 3]
```diff
@@ -554,6 +580,8 @@ done
 
 # Process All partitions From TMPDIR Now
 for partition in ${PARTITIONS}; do
+	foundpart=$(7z l -ba "${FILEPATH}" | gawk '{print $NF}' | grep "${partition}.img")
+	7z e -y -- "${FILEPATH}" "${foundpart}" dummypartition 2>/dev/null
```
Comment (rokibhasansagar) [inline dumper.sh:584]: This part is missing in the `superimage_extract` function, will re-configure this part.

Comment (AndroiableDroid) [reply dumper.sh:584]: Ya that's why added out of all conditions it will work in almost every case I think

Comment (rokibhasansagar) [reply dumper.sh:584]: It was already a part of superimage extraction function, but now it can be used universally.

**Comment by rokibhasansagar** [comment]
Looks Good...
Merging...

**Pull request merged**